pkg/utils/gitrepo.go

```diff
@@ -170,7 +170,7 @@ func getConnectionOptions(cloneOptions *GitCloneOption, primary bool) (connectio
 	if !primary {
 		if cloneOptions.SecondaryConnectionOption == nil {
 			klog.Error("no secondary channel to try")
-			return nil, errors.New("no secondary channel to try")
+			return nil, nil
```

Comment:
I traced thru the code and it seems like the primary getConnectionOptions error is now swallowed and not being returned in at least one logic path.